sonar-plugin-api/src/main/java/org/sonar/api/utils/Version.java

```diff
@@ -93,55 +107,72 @@ public String qualifier() {
    *   <li>1-beta-1</li>
    *   <li>1.2-beta-1</li>
    *   <li>1.2.3-beta-1</li>
+   *   <li>1.2.3.4567</li>
+   *   <li>1.2.3.4567-beta-1</li>
    * </ul>
    * Note that the optional qualifier is the part after the first "-".
    *
    * @throws IllegalArgumentException if parameter is badly formatted, for example
-   * if it defines 4 integer-sequences.
+   * if it defines 5 integer-sequences.
    */
   public static Version parse(String text) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3776)